tensorflow/contrib/gdr/gdr_memory_manager.cc

```diff
@@ -390,92 +358,74 @@ void GdrMemoryManager::TransportOptionsFromTensor(
     ::google::protobuf::Any* mutable_transport_options, const Tensor& tensor,
     Device* device, DeviceContext* device_context, bool on_host,
     StatusCallback done) {
+  ibv_mr* mr = FindMemoryRegion(&tensor);
   auto buffer = DMAHelper::buffer(&tensor);
-  void* addr = buffer->data();
-  size_t length = buffer->size();
-  if (length == 0) {
-    done(errors::Unavailable("Cannot register tensor buffer of size 0"));
-    return;
-  }
-
-  ibv_mr* mr = FindMemoryRegion(addr, length);
 
 #if GOOGLE_CUDA
   if (device->tensorflow_gpu_device_info() && !on_host) {
     Allocator* alloc = GPUProcessState::singleton()->GetCUDAHostAllocator(0);
```

Comment:
OK as-is for now.  Eventually instead of 0 you should use the numa node of either the GPU you're copying from or the NIC that's going to read the CPU buffer.

Reply:
Resolved. I believe we had this discussion a long time ago in #10629. Things certainly changed :P